app-backend/api/src/main/scala/annotation-project/AnnotationProjectPermissionRoutes.scala

```diff
@@ -141,4 +161,65 @@ trait AnnotationProjectPermissionRoutes
       }
     }
   }
+
+  def shareAnnotationProject(projectId: UUID): Route = authenticate { user =>
+    val shareCount =
+      AnnotationProjectDao
+        .getShareCount(projectId, user.id)
+        .transact(xa)
+        .unsafeToFuture
+    authorizeScopeLimit(
+      shareCount,
+      Domain.AnnotationProjects,
+      Action.Share,
+      user
+    ) {
+      entity(as[UserEmail]) { userByEmail =>
+        complete {
+          Auth0Service.getManagementBearerToken flatMap { managementToken =>
+            (for {
+              // Everything has to be Futures here because of methods in akka-http / Auth0Service
+              users <- UserDao
+                .findUsersByEmail(userByEmail.email)
+                .transact(xa)
+                .unsafeToFuture
+              // exist branch:
+              // - share with all users with this email
+              // - update auth0 to make sure those users have annotate access
+              permissions <- users match {
+                case Nil =>
+                  for {
+                    auth0User <- Auth0Service
+                      .createGroundworkUser(userByEmail.email, managementToken)
+                    user <- (auth0User.user_id traverse { userId =>
+                      UserDao.create(
+                        User.Create(
+                          userId,
+                          email = userByEmail.email,
+                          scope = Scopes.GroundworkUser
+                        )
+                      )
+                    }).transact(xa).unsafeToFuture
+                    acrs = user map { getDefaultShare(_) } getOrElse Nil
+                    dbAcrs <- (acrs traverse { acr =>
+                      AnnotationProjectDao
+                        .addPermission(projectId, acr)
+                    }).transact(xa).unsafeToFuture
+                  } yield dbAcrs
+                case us =>
+                  val acrs = getDefaultShare(user)
```

Comment:
Shouldn't this be the `us` (I think this could have a better name like `existingUsers`). Right now this looks like it's updating the user who made the share request and not the users that need to be shared with.

After testing, I've verified that the ACRs added are for the user _making_ the share request, not the users that are intended to be shared with.

```
{TEAM;40bdf3a9-7da0-4b4d-9cfa-918430f67357;ANNOTATE,TEAM;40bdf3a9-7da0-4b4d-9cfa-918430f67357;EDIT,TEAM;40bdf3a9-7da0-4b4d-9cfa-918430f67357;VIEW,TEAM;e36f5155-cd86-43d1-9942-a8eaeba11ca6;ANNOTATE,TEAM;e36f5155-cd86-43d1-9942-a8eaeba11ca6;EDIT,TEAM;e36f5155-cd86-43d1-994
2-a8eaeba11ca6;VIEW,USER;auth0|59318a9d2fbbca3e16bcfc92;VIEW,USER;auth0|59318a9d2fbbca3e16bcfc92;ANNOTATE,USER;auth0|59318a9d2fbbca3e16bcfc92;EXPORT}
```

In this case the dev user is `auth0|59318a9d2fbbca3e16bcfc92`